src/internal/ajax/ajax.ts

```diff
@@ -275,15 +275,53 @@ const LOAD = 'load';
 
 export function fromAjax<T>(config: AjaxConfig): Observable<AjaxResponse<T>> {
   return new Observable((destination) => {
+    // Here we're pulling off each of the configuration arguments
+    // that we don't want to add to the request information we're
+    // passing around.
+    const { params, body: configuredBody, headers: configuredHeaders, ...remainingConfig } = config;
+
+    let { url } = remainingConfig;
+    if (!url) {
+      throw new TypeError('url is required');
+    }
+
+    if (params) {
```

Comment:
Everything inside of this conditional is the bulk of what was added. Please notice that it's mostly comments, and not actually a lot of code. Also worth noticing that `URLSearchParams` is only ever accessed inside of this conditional.